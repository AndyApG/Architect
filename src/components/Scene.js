import React from 'react';
import { connect } from 'react-redux';
import cx from 'classnames';
import PropTypes from 'prop-types';
import { Flipper } from 'react-flip-toolkit';
import { compose, withHandlers } from 'recompose';
import Icon from '@codaco/ui/lib/components/Icon';
import { getActiveProtocolMeta } from '../selectors/protocol';
import { actionCreators as dialogActions } from '../ducks/modules/dialogs';
import history from '../history';
import Start from './Start';
import RecentProtocols from './RecentProtocols';
import Overview from './Overview';
import Timeline from './Timeline';
import ProtocolControlBar from './ProtocolControlBar';
import Screens from './Screens';
import networkCanvasBrand from '../images/network-canvas-brand.svg';

const Scene = ({
  protocolMeta,
  handleClickStart,
}) => {
  const protocolId = protocolMeta && protocolMeta.id;
  const flipKey = protocolId || 'start';
  const showProtocol = !!protocolId;
  const showStart = !protocolId;

  const sceneClasses = cx(
    'scene',
    { 'scene--protocol': showProtocol },
  );

  return (
    <div className={sceneClasses}>
      <div className="scene__background scene__background--top" />
      <div className="scene__background scene__background--bottom" />
      <img className="scene__brand" src={networkCanvasBrand} alt="" />

      <div className="scene__home">
        <Icon onClick={handleClickStart} className="start-button__arrow" name="back-arrow" />
      </div>

      <Flipper flipKey={flipKey}>

        <div className="scene__start">
          <Start show={showStart} />
        </div>

        <div className="scene__recent-protocols">
          <RecentProtocols show={showStart} />
        </div>

        <div className="scene__protocol">
          <Overview
            show={showProtocol}
            flipId={protocolMeta && encodeURIComponent(protocolMeta.filePath)}
          />

          <div className="scene__timeline">
            <Timeline show={showProtocol} />
          </div>

          <ProtocolControlBar show={showProtocol} />
        </div>
      </Flipper>

      <Screens />
    </div>
  );
};

Scene.propTypes = {
  protocolMeta: PropTypes.object,
  handleClickStart: PropTypes.func.isRequired,
};

Scene.defaultProps = {
  protocolMeta: null,
};

const mapStateToProps = state => ({
  hasUnsavedChanges: (state.session.lastChanged > state.session.lastSaved),
  protocolMeta: getActiveProtocolMeta(state),
});

const mapDispatchToProps = {
  openDialog: dialogActions.openDialog,
};

const linkHandler = withHandlers({
  handleClickStart: ({
    hasUnsavedChanges,
    openDialog,
  }) =>
    () => {
      const goToStartScreen = () => history.push('/');

      if (!hasUnsavedChanges) {
        goToStartScreen();
        return;
      }

      openDialog({
        type: 'Warning',
        title: 'Unsaved changes',
        message: (
          <div>
            Are you sure you want to go back to the start screen?
            <p><strong>Any unsaved changes will be lost!</strong></p>
          </div>
        ),
        confirmLabel: 'Go to start screen',
        onConfirm: goToStartScreen,
      });
    },
});

export { Scene };

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  linkHandler,
)(Scene);
