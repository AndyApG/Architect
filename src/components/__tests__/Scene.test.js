/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { Scene } from '../Scene';

const mockProps = {
  protocolMeta: {},
  location: {},
  handleClickStart: () => {},
};

describe('<Scene />', () => {
  it('can render', () => {
    const component = shallow(<Scene {...mockProps} />);

    expect(component.hasClass('scene--protocol')).toBe(false);
    expect(component).toMatchSnapshot();
  });

  it('shows protocol when there is one', () => {
    const props = { protocolMeta: { id: 'foo' } };
    const component = shallow(<Scene {...mockProps} {...props} />);

    expect(component.hasClass('scene--protocol')).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
