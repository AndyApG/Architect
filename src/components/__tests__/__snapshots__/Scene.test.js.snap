// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Scene /> can render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scene
    location={Object {}}
    protocolMeta={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="scene__background scene__background--top"
        />,
        <div
          className="scene__background scene__background--bottom"
        />,
        <img
          alt=""
          className="scene__brand"
          src="network-canvas-brand.svg"
        />,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="scene__home"
          exact={true}
          to="/"
        >
          <img
            alt=""
            src="architect-logo-icon.svg"
          />
        </NavLink>,
        <Flipper
          applyTransformOrigin={true}
          element="div"
          flipKey="start"
        >
          <div
            className="scene__start"
          >
            <Connect(Start)
              show={true}
            />
          </div>
          <div
            className="scene__recent-protocols"
          >
            <Connect(RecentProtocols)
              show={true}
            />
          </div>
          <div
            className="scene__protocol"
          >
            <Connect(Overview)
              show={false}
            />
            <div
              className="scene__timeline"
            >
              <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                show={false}
              />
            </div>
            <Connect(ProtocolControlBar)
              show={false}
            />
          </div>
        </Flipper>,
        <Connect(Cards)
          location={Object {}}
        />,
      ],
      "className": "scene",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "scene__background scene__background--top",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "scene__background scene__background--bottom",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "scene__brand",
          "src": "network-canvas-brand.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <img
            alt=""
            src="architect-logo-icon.svg"
          />,
          "className": "scene__home",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "architect-logo-icon.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "applyTransformOrigin": true,
          "children": Array [
            <div
              className="scene__start"
            >
              <Connect(Start)
                show={true}
              />
            </div>,
            <div
              className="scene__recent-protocols"
            >
              <Connect(RecentProtocols)
                show={true}
              />
            </div>,
            <div
              className="scene__protocol"
            >
              <Connect(Overview)
                show={false}
              />
              <div
                className="scene__timeline"
              >
                <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                  show={false}
                />
              </div>
              <Connect(ProtocolControlBar)
                show={false}
              />
            </div>,
          ],
          "element": "div",
          "flipKey": "start",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Start)
                show={true}
              />,
              "className": "scene__start",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "show": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(RecentProtocols)
                show={true}
              />,
              "className": "scene__recent-protocols",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "show": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Overview)
                  show={false}
                />,
                <div
                  className="scene__timeline"
                >
                  <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                    show={false}
                  />
                </div>,
                <Connect(ProtocolControlBar)
                  show={false}
                />,
              ],
              "className": "scene__protocol",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "flipId": undefined,
                  "show": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                    show={false}
                  />,
                  "className": "scene__timeline",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "show": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "show": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="scene__background scene__background--top"
          />,
          <div
            className="scene__background scene__background--bottom"
          />,
          <img
            alt=""
            className="scene__brand"
            src="network-canvas-brand.svg"
          />,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="scene__home"
            exact={true}
            to="/"
          >
            <img
              alt=""
              src="architect-logo-icon.svg"
            />
          </NavLink>,
          <Flipper
            applyTransformOrigin={true}
            element="div"
            flipKey="start"
          >
            <div
              className="scene__start"
            >
              <Connect(Start)
                show={true}
              />
            </div>
            <div
              className="scene__recent-protocols"
            >
              <Connect(RecentProtocols)
                show={true}
              />
            </div>
            <div
              className="scene__protocol"
            >
              <Connect(Overview)
                show={false}
              />
              <div
                className="scene__timeline"
              >
                <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                  show={false}
                />
              </div>
              <Connect(ProtocolControlBar)
                show={false}
              />
            </div>
          </Flipper>,
          <Connect(Cards)
            location={Object {}}
          />,
        ],
        "className": "scene",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "scene__background scene__background--top",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "scene__background scene__background--bottom",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "scene__brand",
            "src": "network-canvas-brand.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt=""
              src="architect-logo-icon.svg"
            />,
            "className": "scene__home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "architect-logo-icon.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "applyTransformOrigin": true,
            "children": Array [
              <div
                className="scene__start"
              >
                <Connect(Start)
                  show={true}
                />
              </div>,
              <div
                className="scene__recent-protocols"
              >
                <Connect(RecentProtocols)
                  show={true}
                />
              </div>,
              <div
                className="scene__protocol"
              >
                <Connect(Overview)
                  show={false}
                />
                <div
                  className="scene__timeline"
                >
                  <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                    show={false}
                  />
                </div>
                <Connect(ProtocolControlBar)
                  show={false}
                />
              </div>,
            ],
            "element": "div",
            "flipKey": "start",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Start)
                  show={true}
                />,
                "className": "scene__start",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "show": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(RecentProtocols)
                  show={true}
                />,
                "className": "scene__recent-protocols",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "show": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(Overview)
                    show={false}
                  />,
                  <div
                    className="scene__timeline"
                  >
                    <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                      show={false}
                    />
                  </div>,
                  <Connect(ProtocolControlBar)
                    show={false}
                  />,
                ],
                "className": "scene__protocol",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "flipId": undefined,
                    "show": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                      show={false}
                    />,
                    "className": "scene__timeline",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "show": false,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "show": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Scene /> shows protocol when there is one 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scene
    location={Object {}}
    protocolMeta={
      Object {
        "id": "foo",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="scene__background scene__background--top"
        />,
        <div
          className="scene__background scene__background--bottom"
        />,
        <img
          alt=""
          className="scene__brand"
          src="network-canvas-brand.svg"
        />,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="scene__home"
          exact={true}
          to="/"
        >
          <img
            alt=""
            src="architect-logo-icon.svg"
          />
        </NavLink>,
        <Flipper
          applyTransformOrigin={true}
          element="div"
          flipKey="foo"
        >
          <div
            className="scene__start"
          >
            <Connect(Start)
              show={false}
            />
          </div>
          <div
            className="scene__recent-protocols"
          >
            <Connect(RecentProtocols)
              show={false}
            />
          </div>
          <div
            className="scene__protocol"
          >
            <Connect(Overview)
              show={true}
            />
            <div
              className="scene__timeline"
            >
              <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                show={true}
              />
            </div>
            <Connect(ProtocolControlBar)
              show={true}
            />
          </div>
        </Flipper>,
        <Connect(Cards)
          location={Object {}}
        />,
      ],
      "className": "scene scene--protocol",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "scene__background scene__background--top",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "scene__background scene__background--bottom",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "scene__brand",
          "src": "network-canvas-brand.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <img
            alt=""
            src="architect-logo-icon.svg"
          />,
          "className": "scene__home",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "architect-logo-icon.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "applyTransformOrigin": true,
          "children": Array [
            <div
              className="scene__start"
            >
              <Connect(Start)
                show={false}
              />
            </div>,
            <div
              className="scene__recent-protocols"
            >
              <Connect(RecentProtocols)
                show={false}
              />
            </div>,
            <div
              className="scene__protocol"
            >
              <Connect(Overview)
                show={true}
              />
              <div
                className="scene__timeline"
              >
                <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                  show={true}
                />
              </div>
              <Connect(ProtocolControlBar)
                show={true}
              />
            </div>,
          ],
          "element": "div",
          "flipKey": "foo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Start)
                show={false}
              />,
              "className": "scene__start",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "show": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(RecentProtocols)
                show={false}
              />,
              "className": "scene__recent-protocols",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "show": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Overview)
                  show={true}
                />,
                <div
                  className="scene__timeline"
                >
                  <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                    show={true}
                  />
                </div>,
                <Connect(ProtocolControlBar)
                  show={true}
                />,
              ],
              "className": "scene__protocol",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "flipId": undefined,
                  "show": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                    show={true}
                  />,
                  "className": "scene__timeline",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "show": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "show": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="scene__background scene__background--top"
          />,
          <div
            className="scene__background scene__background--bottom"
          />,
          <img
            alt=""
            className="scene__brand"
            src="network-canvas-brand.svg"
          />,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="scene__home"
            exact={true}
            to="/"
          >
            <img
              alt=""
              src="architect-logo-icon.svg"
            />
          </NavLink>,
          <Flipper
            applyTransformOrigin={true}
            element="div"
            flipKey="foo"
          >
            <div
              className="scene__start"
            >
              <Connect(Start)
                show={false}
              />
            </div>
            <div
              className="scene__recent-protocols"
            >
              <Connect(RecentProtocols)
                show={false}
              />
            </div>
            <div
              className="scene__protocol"
            >
              <Connect(Overview)
                show={true}
              />
              <div
                className="scene__timeline"
              >
                <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                  show={true}
                />
              </div>
              <Connect(ProtocolControlBar)
                show={true}
              />
            </div>
          </Flipper>,
          <Connect(Cards)
            location={Object {}}
          />,
        ],
        "className": "scene scene--protocol",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "scene__background scene__background--top",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "scene__background scene__background--bottom",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "scene__brand",
            "src": "network-canvas-brand.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt=""
              src="architect-logo-icon.svg"
            />,
            "className": "scene__home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "architect-logo-icon.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "applyTransformOrigin": true,
            "children": Array [
              <div
                className="scene__start"
              >
                <Connect(Start)
                  show={false}
                />
              </div>,
              <div
                className="scene__recent-protocols"
              >
                <Connect(RecentProtocols)
                  show={false}
                />
              </div>,
              <div
                className="scene__protocol"
              >
                <Connect(Overview)
                  show={true}
                />
                <div
                  className="scene__timeline"
                >
                  <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                    show={true}
                  />
                </div>
                <Connect(ProtocolControlBar)
                  show={true}
                />
              </div>,
            ],
            "element": "div",
            "flipKey": "foo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Start)
                  show={false}
                />,
                "className": "scene__start",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "show": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(RecentProtocols)
                  show={false}
                />,
                "className": "scene__recent-protocols",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "show": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(Overview)
                    show={true}
                  />,
                  <div
                    className="scene__timeline"
                  >
                    <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                      show={true}
                    />
                  </div>,
                  <Connect(ProtocolControlBar)
                    show={true}
                  />,
                ],
                "className": "scene__protocol",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "flipId": undefined,
                    "show": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <withRouter(withStateHandlers(defaultProps(Connect(sortableList(Timeline)))))
                      show={true}
                    />,
                    "className": "scene__timeline",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "show": true,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "show": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
